src/brushes/circle_brush.class.js

```diff
@@ -76,6 +76,9 @@ fabric.CircleBrush = fabric.util.createClass(fabric.BaseBrush, /** @lends fabric
       this._render();
     }
     else {
+      if (this.limitedToCanvasSize === true && this._isOutSideCanvas(pointer)) {
+        return;
+      }
```

Comment:
in the pencil brush we are returning immediately if the condition match.
Why are we allowing the full render here?